lib/cli/src/commands/create_exe.rs

```diff
@@ -1482,142 +1526,123 @@ mod http_fetch {
-                            }
-                        }
-                    }
-                    Err(err) => {
-                        eprintln!(
-                            "Could not determine cache path for downloaded binaries.: {}",
-                            err
-                        );
-                    }
-                }
-                let _response = download_thread
-                    .join()
-                    .expect("Could not join downloading thread");
-                match super::get_libwasmer_cache_path() {
-                    Ok(mut cache_path) => {
-                        cache_path.push(&filename);
-                        if !cache_path.exists() {
-                            if let Err(err) = std::fs::copy(&filename, &cache_path) {
-                                eprintln!(
-                                    "Could not store tarball to cache path `{}`: {}",
-                                    cache_path.display(),
-                                    err
-                                );
-                            } else {
-                                eprintln!(
-                                    "Cached tarball to cache path `{}`.",
-                                    cache_path.display()
-                                );
-                            }
-                        }
-                    }
-                    Err(err) => {
-                        eprintln!(
-                            "Could not determine cache path for downloaded binaries.: {}",
-                            err
-                        );
-                    }
+        if assets.len() != 1 {
+            return Err(anyhow!(
+                "GitHub API: more that one release selected for target {target_triple}: {assets:?}"
+            ));
+        }
+
+        let browser_download_url = if let Some(url) = assets[0]["browser_download_url"].as_str() {
+            url.to_string()
+        } else {
+            return Err(anyhow!(
+                "Could not get download url from Github API response."
+            ));
+        };
+
+        let filename = browser_download_url
+            .split('/')
+            .last()
+            .unwrap_or("output")
+            .to_string();
+
+        let download_tempdir = tempdir::TempDir::new("wasmer-download")?;
+        let download_path = download_tempdir.path().to_path_buf().join(&filename);
+
+        let mut file = std::fs::File::create(&download_path)?;
+        eprintln!(
+            "Downloading {} to {}",
+            browser_download_url,
+            download_path.display()
+        );
+
+        let uri = Uri::try_from(browser_download_url.as_str())?;
+        let mut response = Request::new(&uri)
+            .header("User-Agent", "wasmer")
+            .send(&mut file)
+            .map_err(anyhow::Error::new)
+            .context("Could not lookup wasmer artifact on Github.")?;
+        if response.status_code() == StatusCode::new(302) {
```

Comment:
It might be better to check against the `StatusCode::FOUND` constant rather than the number.

We can also tell `reqwest` to follow redirects automatically with [`ClientBuilder::redirect()`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.redirect) and drop the explicit redirect handling. By default, it will follow up to 10 steps before giving up.

Reply:
migrated to reqwest